CMakeLists.txt

```diff
@@ -126,6 +126,26 @@ option(IREE_ENABLE_LLD "Use lld when linking" OFF)
 option(IREE_ENABLE_ASAN "Enable address sanitizer" OFF)
 option(IREE_ENABLE_MSAN "Enable memory sanitizer" OFF)
 option(IREE_ENABLE_TSAN "Enable thread sanitizer" OFF)
+option(IREE_BYTECODE_MODULE_ENABLE_TSAN "Enable thread sanitizer in IREE modules in tests" OFF)
+
+if(NOT IREE_ENABLE_TSAN STREQUAL IREE_BYTECODE_MODULE_ENABLE_TSAN)
```

Comment:
I think it's also worth a brief comment here about why we're doing it this way (as opposed to having one flag).

Reply:
I dropped the `if(IREE_BUILD_TESTS)` because I realized that the actual relevant condition here is "are we building any `iree_bytecode_module`) and that's not quite the same? Would you know how to express that --- is this some OR combination of things like IREE_BUILD_TESTS, IREE_BUILD_SAMPLES, ... ?

Re `STREQUAL` - do you mean I should be able to use `EQUAL` here ?  I dont really understand that that does, sorry. Is it "equality comparison after implict conversion to bool" ?

Reply:
I think it's really "Are we building and then executing any iree_bytecode_module". We could be building it for use on some other platform (or could be executing one we built previously). If these options literally always have to be the same then there's not much point in having them :grin: I'm alright if you want to leave it like this for now until we can figure out the exact right condition.

Branching STREQUAL to a separate thread since it's really a separate question

Reply:
> If these options literally always have to be the same then there's not much point in having them

I guess there's 2 reasons to have them:
* For the sake of "options that do one thing each". I have in the past proposed options doing multiple things and that was turned down on that basis :slightly_smiling_face: 
* In case we would some day figure a clear condition for "Are we building and then executing any iree_bytecode_module" so we could reenclose that if into an if on that condition, like we had before with that `if(IREE_BUILD_TESTS)` that was dropped.  In that sense, these options don't quite "literally always have to be the same", it may be just a temporary state of things.

Reply:
Yeah, the latter thing is what I was referring to. SGTM :-)